messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ElytronSecurityManager.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.messaging.activemq;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.activemq.artemis.core.security.CheckType;
+import org.apache.activemq.artemis.core.security.Role;
+import org.apache.activemq.artemis.spi.core.security.ActiveMQSecurityManager;
+import org.wildfly.extension.messaging.activemq.logging.MessagingLogger;
+import org.wildfly.security.auth.server.RealmUnavailableException;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.auth.server.ServerAuthenticationContext;
+import org.wildfly.security.evidence.PasswordGuessEvidence;
+
+/**
+ * This class implements an {@link ActiveMQSecurityManager} that uses an Elytron {@link SecurityDomain} to authenticate
+ * users and perform role checking.
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class ElytronSecurityManager implements ActiveMQSecurityManager {
+
+    private final SecurityDomain securityDomain;
+    private final String defaultUser;
+    private final String defaultPassword;
+
+    /**
+     * Creates an instance of {@link ElytronSecurityManager} with the specified {@link SecurityDomain}.
+     *
+     * @param securityDomain a reference to the Elytron {@link SecurityDomain} that will be used to authenticate users.
+     */
+    public ElytronSecurityManager(final SecurityDomain securityDomain) {
+        if (securityDomain == null)
+            throw MessagingLogger.ROOT_LOGGER.invalidNullSecurityDomain();
+        this.securityDomain = securityDomain;
+        this.defaultUser = DefaultCredentials.getUsername();
+        this.defaultPassword = DefaultCredentials.getPassword();
+    }
+
+    @Override
+    public boolean validateUser(String username, String password) {
+
+        if (defaultUser.equals(username) && defaultPassword.equals(password))
+            return true;
+
+        ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+        PasswordGuessEvidence evidence = new PasswordGuessEvidence(password != null ? password.toCharArray() : null);
+        try {
+            context.setAuthenticationName(username);
+            if (context.verifyEvidence(evidence)) {
+                return true;
```

Comment:
What does this 'true' mean - I think this needs the authorize() call as in the next method as that is where the LoginPermission is checked.

Where the accounts of identities are enabled / disabled one way this could be solved it be changing whether that identity is granted the LoginPermission so I think the check will be important here.